### ISSUE TITLE
exporting scraped PDF files

### ISSUE THREAD

**Issue by dsanson** [jawj/Zot2Bib#1]
I'd love to see support for exporting scraped PDF files. I poked around in the code a bit but couldn't make sense of what might need to be done, but I really don't know anything about the inner workings of Zotero.

Is it a matter of passing the `exportFileData` option when calling the bibtex.js translator? Or is it a matter of setting an increased delay, so that the export to bibtex occurs after zotero has scraped the PDF?

**Comment by vancleve** [comment]
Just a quick *bump* on this issue since I've updated my fork again. It now continues to export the bibtex even when the PDF download fails.

**Comment by foice** [comment]
Well, I am considering using Zotero in placeo of BibDesk and I have to say that having the export of the bibliographic info, as it currently works, plus the location of the PDF file in the BiBDesk info would be awesome.

As far as I can see zot2bib is triggered on newly imported items and cannot be triggered to repeat the export, e.g. after that the PDF file has been fetched. Am I missing something?

I think that in general it is useful to be able to repeat (and update or overwrite) the addition to bibdesk, for instance for any zotero item that I have updated.

**Comment by nathan-artist** [comment]
@foice: If you plan to use Zotero _in place of_ BibDesk (as opposed to _with_ BibDesk), why not use [Better BibTeX for Zotero](https://github.com/retorquere/zotero-better-bibtex)? If you really plan to replace BibDesk with Zotero, you wouldn't need BibDesk at all.

**Comment by foice** [comment]
I have [https://github.com/retorquere/zotero-better-bibtex](https://github.com/retorquere/zotero-better-bibtex) currently "on trial". The most likely outcome seems to be I have to use it _with_ Bibdeksk, because so far I can handle addition via command line only with Bibdesk, plus there are another number of "on the field" tests I have not made yet on Zotero. Still Zotero gives probably better keywords features (for my use). 

Anyhow, this is all about "me". On the contrary I think the issue of zot2bib being triggerable at will is an issue with its own standing ... regardless of my usecase. So I restate the case for having zot2bib to be called on already existing items, why should it not be possible?

At any rate, exporting the position of the PDF file to bibdesk own field seems also a core feature.

**Comment by nathan-artist** [comment]
@foice: Yes, that is a major limitation of Better BibTeX for Zotero: changes to the BibTeX are one-way from Zotero. One has to close a BibDesk database before editing the BibTeX outside of BibDesk, but at least it is possible to edit it under that condition.

You may want to open a separate issue for "having zot2bib to be called on already existing items", which seems out of scope for this issue.

**Comment by vancleve** [comment]
Just FYI, I moved fully over to Zotero in 2019 because Bibdesk was just too slow with my big library. Zotero is quite a bit harder add functionality to, compared to the script hooks in Bibdesk, but its much more actively developed than Bibdesk.

**Comment by nathan-artist** [comment]
@vancleve: I don't know what slowed down your BibDesk library, but in my case I figured out quickly when I started using BibDesk in 2008 that the linked file fields were slowing it down, and that using them was not going to scale to the many thousands of references that I foresaw having in the near future. Instead, I made a "Downloaded" checkbox field and an AppleScript that opens the downloaded file without any use of the linked file field. Now I have well over 30k references in one BibDesk database and it is lightning fast, much faster than Zotero.

Having said that, there are some amazing plugins for Zotero that shouldn't be ignored, and by using Zotero in place of BibDesk you have easy access to those plugins.

**Comment by vancleve** [comment]
@nathan-artist, ah yes, that was probably it. I had over 10k refs with linked PDFs and it was terribly slow.

Still, there is some elegance to cutting out the zot2bib middle man and mostly just using one reference manager instead of two, even if Zotero is a bit slow.
